layers/gd/Dockerfile

```diff
@@ -0,0 +1,55 @@
+ARG PHP_VERSION
+FROM bref/build-php-$PHP_VERSION as ext
+
+# Install required build libraries
+RUN set -ue \
+    ; LD_LIBRARY_PATH= \
+    ; yum -y install \
+        libwebp-devel \
+        libXpm-devel \
+        libpng-devel \
+        libjpeg-devel \
+        freetype-devel \
+    ; yum clean all \
+;
+
+WORKDIR ${PHP_BUILD_DIR}/ext/gd
+RUN phpize
+ARG PHP_VERSION
+RUN if [ "$PHP_VERSION" = "73" ] ; \
+    then \
+        ./configure \
+        --disable-static \
+        --with-gd \
+        --enable-gd-jis-conv \
+        --enable-shared \
+        --with-zlib-dir=${INSTALL_DIR} \
+        --with-jpeg-dir=${INSTALL_DIR} \
+        --with-png-dir=${INSTALL_DIR} \
+        --with-webp-dir=${INSTALL_DIR} \
+        --with-xpm-dir=${INSTALL_DIR} \
+        --with-freetype-dir=${INSTALL_DIR} \
+        ; \
+    else \
+        ./configure \
+        --disable-static \
+        --enable-gd \
+        --enable-gd-jis-conv \
+        --enable-shared \
+        --with-freetype \
+        --with-jpeg \
+        --with-webp \
+        --with-xpm \
+        ; \
+    fi \
+;
+RUN make -j $(nproc)
+RUN make install
+RUN cp "$(php-config --extension-dir)/gd.so" /tmp/gd.so
+
+FROM scratch
+
+COPY --from=ext /usr/lib64/libXpm.so.4 /opt/bref/lib64/libXpm.so.4
+COPY --from=ext /usr/lib64/libwebp.so.4 /opt/bref/lib64/libwebp.so.4
```

Comment:
Regarding [Amazon Linux 1 packages](https://github.com/lambci/docker-lambda/blob/master/base/packages.txt), we have only the 2 following ones to copy.

Reply:
I'd copy the exact version we built against .. not relying on sys libs that could break something ...

Reply:
I'm not sure to understand what you mean. If it's about copying ALL the shared libraries I'm not agree at all because as you can see with `ldd` on gd.so: 
```
        linux-vdso.so.1 =>  (0x00007fff11fb3000)
        libX11.so.6 => /usr/lib64/libX11.so.6 (0x00007f835e7f4000)
        libXpm.so.4 => /opt/bref/lib64/libXpm.so.4 (0x00007f835e5e3000)
        libpng12.so.0 => /usr/lib64/libpng12.so.0 (0x00007f835e3bd000)
        libz.so.1 => /opt/bref/lib/libz.so.1 (0x00007f835e1a0000)
        libjpeg.so.62 => /usr/lib64/libjpeg.so.62 (0x00007f835df4d000)
        libwebp.so.4 => /opt/bref/lib64/libwebp.so.4 (0x00007f835dcfe000)
        libfreetype.so.6 => /usr/lib64/libfreetype.so.6 (0x00007f835da62000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f835d694000)
        libxcb.so.1 => /usr/lib64/libxcb.so.1 (0x00007f835d476000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f835d272000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f835cf70000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f835cd54000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f835ed9a000)
        libXau.so.6 => /usr/lib64/libXau.so.6 (0x00007f835cb51000)
```

We should in this case even copy the libc, libdl and other ones and we would probably collide with every other extensions because every of them have links to those libs.

Amazon don't change their base images very often but yes it breaks ABI when it's the case and it should be managed before it happen.